File: SpinEditor/CustomDisplayText/CustomDisplayTextVB/MyRadSpinEditor.vb
Imports Telerik.WinControls.UI

#Region "CustomRadSpinEditor"
Public Class MyRadSpinEditor
    Inherits RadSpinEditor

    Private spinElement As MyRadSpinElement

    Public Overrides Property ThemeClassName As String
        Get
            Return GetType(RadSpinEditor).FullName
        End Get
        Set(value As String)
            MyBase.ThemeClassName = value
        End Set
    End Property
    Public Property ScientificNatation As Boolean
        Get
            Return Me.spinElement.ScientificNation
        End Get
        Set(ByVal value As Boolean)
            Me.spinElement.ScientificNation = value
        End Set
    End Property

    Public Property LeadingZero As Boolean
        Get
            Return Me.spinElement.LeadingZero
        End Get
        Set(ByVal value As Boolean)
            Me.spinElement.LeadingZero = value
        End Set
    End Property

    Protected Overrides Function CreateSpinElement() As RadSpinElement
        Me.spinElement = New MyRadSpinElement()
        Return Me.spinElement
    End Function
End Class
#End Region

File: SpinEditor/CustomDisplayText/CustomDisplayText/MyRadSpinEditor.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Telerik.WinControls.UI;

namespace CustomDisplayText
{
    #region CustomRadSpinEditor
    public class MyRadSpinEditor : RadSpinEditor
    {
        private MyRadSpinElement spinElement;

        public override string ThemeClassName
        {
            get
            {
                return typeof(RadSpinEditor).FullName;
            }
        }

        public bool ScientificNatation
        {
            get
            {
                return this.spinElement.ScientificNation;
            }
            set
            {
                this.spinElement.ScientificNation = value;
            }
        }

        public bool LeadingZero
        {
            get
            {
                return this.spinElement.LeadingZero;
            }
            set
            {
                this.spinElement.LeadingZero = value;
            }
        }

        protected override RadSpinElement CreateSpinElement()
        {
            this.spinElement = new MyRadSpinElement();

            return this.spinElement;
        }
    }
    #endregion
}
